app/views/govuk_publishing_components/components/docs/autocomplete.yml

```diff
@@ -0,0 +1,70 @@
+
+name: Autocomplete (experimental)
+description: The autocomplete component lets users select from a dynamic list of suggestions to populate the value of a text input element.
+govuk_frontend_components:
+  - autocomplete
+body: |
+  If suggestions come from a static source then enchancing a server rendered HTML select element is a better option.
+
+  This component is currently experimental because more research is needed to validate it pending autocomplete implementation documented in the [GDS roadmap](https://design-system.service.gov.uk/community/upcoming-components-patterns/).  If using this component, [please feed back any research findings to the Design System team](https://design-system.service.gov.uk/get-in-touch/).
+accessibility_criteria: |
+  * Be focusable with a keyboard
+  * Indicate when it has keyboard focus
+  * Inform the user that it is an editable field
+  * Inform the user if there is a pre-filled value
+  * Inform the user that autocomplete is available
+  * Explain how to use autocomplete
+  * Inform the user that content has been expanded
+  * Inform the user when there are matches, or if there are no matches
+  * Inform the user as the number of matches changes
+  * Enable the user to navigate the available matches using touch or keyboard
+  * Inform the user when a match is selected
+  * Inform the user if a match is pre-selected
+  * Enable the user to confirm the selected match
+  * Inform the user when a match is confirmed
+  * Return focus to the editable field when a selected match is confirmed
+uses_component_wrapper_helper: true
+shared_accessibility_criteria:
+  - link
+examples:
+  default:
+    description: url to get suggestion data from or an array of suggestions (required). Query parameter of ?q={inputValue} will be appended The array option exists mainly for the purposes of demonstration and testing where an HTTP endpoint is not ideal. If you have a static array of suggestions consider another approach that enhances an HTML select element.
+    data:
+      source: ['prime minister','deputy prime minister','contact prime minister','email prime minister','last prime minister']
+  custom_input_element:
```

Comment:
Should this be an option on the component? It seems like it would be possible to pass anything here, even something like a simple paragraph - that wouldn't be then rendered as an autocomplete. I suspect you've had to do extra work to make this option possible. 

I'd recommend not doing that (it feels inherently brittle, as presumably your JS has to assume certain things about what might be in this block) but instead give the component options to display whatever is required here - the example talks about a custom input element, what are the requirements? e.g. if it's to be able to set the label text, or the input id, or having the search button at the end - make those options that get passed to the component instead.

Reply:
I've considered this and I don't think allowing a block to be passed to this component makes any sense. If there's a requirement to have different input types I think that should be controlled through options, not through passing a block - there's too much risk involved.

Reply:
ok - its no bother to remove the block option. if this component gets taken forward in the future and it becomes a requirement, someone can implement it in a better thought out way than maybe i have :D